yt_dlp/extractor/slideslive.py

```diff
@@ -369,15 +398,47 @@ def _extract_custom_m3u8_info(self, m3u8_data):
 
         return m3u8_dict
 
-    def _extract_formats(self, cdn_hostname, path, video_id):
-        formats = []
-        formats.extend(self._extract_m3u8_formats(
+    def _parse_m3u8_vod_duration(self, m3u8_vod, video_id):
+        # SlidesLive VOD m3u8 does not contain '#EXT-X-PLAYLIST-TYPE:VOD'
+        if '#EXT-X-ENDLIST' not in m3u8_vod:
```

Comment:
I'm also wondering if this is a change that should be made to `InfoExtractor._parse_m3u8_vod_duration` instead of overriding it here. Duration can be determined if the m3u8 contains `#EXT-X-ENDLIST`, but not all VOD m3u8s contain `#EXT-X-PLAYLIST-TYPE:VOD`

Reply:
Makes sense to do in core